lib/pure/os.nim

```diff
@@ -865,6 +865,19 @@ type
 
 {.deprecated: [TPathComponent: PathComponent].}
 
+
+when defined(posix):
+  proc getSymlinkFileKind(path: string): PathComponent =
+    # Helper function.
+    var s : Stat
```

Comment:
Is the additional check really necessary? It's only called with a normal path and the assert prevent other calls. I could add an assert if you like.